types/API.d.ts

```diff
@@ -0,0 +1,487 @@
+   * Uses human-friendly suffixes: ‘K’ for kilobytes, ‘M’ for megabytes, ‘G’ for gigabytes’, etc. Eg “150M”.
+   */
+  max_memory_restart?: number | string;
+  /**
+   * Arguments to pass to the interpreter
+   */
+  node_args?: string | string[];
+  /**
+   * Prefix logs with time
+   */
+  time?: boolean;
+  /**
+   * This will make PM2 listen for that event. In your application you will need to add process.send('ready');
+   * when you want your application to be considered as ready.
+   */
+  wait_ready?: boolean;
+  /**
+   * (Default: 1600)
+   * The number of milliseconds to wait after a stop or restart command issues a SIGINT signal to kill the
+   * script forceably with a SIGKILL signal.
+   */
+  kill_timeout?: number;
+  /**
+   * (Default: 0) Number of millseconds to wait before restarting a script that has exited.
+   */
+  restart_delay?: number;
+  /**
+   * (Default: “node”) The interpreter for your script (eg “python”, “ruby”, “bash”, etc).
+   * The value “none” will execute the ‘script’ as a binary executable.
+   */
+  interpreter?: string;
+  /**
+   * (Default: ‘fork’) If sets to ‘cluster’, will enable clustering
+   * (running multiple instances of the script).
+   */
+  exec_mode?: string;
+  /**
+   * (Default: 1) How many instances of script to create. Only relevant in exec_mode ‘cluster’.
+   */
+  instances?: number;
+  /**
+   * (Default: false) If true, merges the log files for all instances of script into one stderr log
+   * and one stdout log. Only applies in ‘cluster’ mode. For example, if you have 4 instances of
+   * ‘test.js’ started via pm2, normally you would have 4 stdout log files and 4 stderr log files,
+   * but with this option set to true you would only have one stdout file and one stderr file.
+   */
+  merge_logs?: boolean;
+  /**
+   * If set to true, the application will be restarted on change of the script file.
+   */
+  watch?: boolean|string[];
+  /**
+   * (Default: false) By default, pm2 will only start a script if that script isn’t
+   * already running (a script is a path to an application, not the name of an application
+   * already running). If force is set to true, pm2 will start a new instance of that script.
+   */
+  force?: boolean;
+  ignore_watch?: string[];
+  cron?: any;
+  execute_command?: any;
+  write?: any;
+  source_map_support?: any;
+  disable_source_map_support?: any;
+  /**
+   * The environment variables to pass on to the process.
+   */
+  env?: { [key: string]: string; };
+}
+
+interface ReloadOptions {
+  /**
+   * (Default: false) If true is passed in, pm2 will reload it’s environment from process.env
+   * before reloading your process.
+   */
+  updateEnv?: boolean;
+}
+
+// Types
+
+type ProcessStatus = 'online' | 'stopping' | 'stopped' | 'launching' | 'errored' | 'one-launch-status';
+type Platform = 'ubuntu' | 'centos' | 'redhat' | 'gentoo' | 'systemd' | 'darwin' | 'amazon';
+
+type ErrCallback = (err: Error) => void;
```

Comment:
Don't you need `Error | null` here too? `connect` uses this and won't always return an error.

Reply:
Out of scope for this PR, as that's the already existing typings iirc

Reply:
It is the existing type but you've changed the ones below anyway so may as well do this one.

Existing typings:
https://github.com/Unitech/pm2/blob/2573516e9321a78fb10474ea58c2cb487a663de6/types/index.d.ts#L448-L453